library/src/scala/runtime/DynamicTuple.scala

```diff
@@ -253,6 +249,18 @@ object DynamicTuple {
     res.asInstanceOf[Result]
   }
 
+  def dynamicZip[This <: Tuple, T2 <: Tuple](t1: This, t2: T2): Zip[This, T2] = {
+    if (t1.size == 0 || t2.size == 0) ().asInstanceOf[Zip[This, T2]]
+    else Tuple.fromArray(
+      t1.asInstanceOf[Product].productIterator.zip(
+      t2.asInstanceOf[Product].productIterator).toArray
+    ).asInstanceOf[Zip[This, T2]]
+  }
+
+  def dynamicMap[This <: Tuple, F[_]](self: This, f: [t] => t => F[t]): Map[This, F] =
+    Tuple.fromArray(self.asInstanceOf[Product].productIterator.map(f(_)).toArray)
```

Comment:
```suggestion
    Tuple.fromArray(self.asInstanceOf[Product].productIterator.map(f(_)).toArray) // TODO use toIArray of Object to avoid double/triple array copy
```

Reply:
Can you clarify on what this change exactly will do and why can't we have it now?

Reply:
Because we still haven't implemented `toIArray`. If you look follow the implementation of `fromArray` you will see several places where the array has to be copied. In `fromIArray` this is reduced to one case that we can avoid.